images/macos/provision/core/openjdk.sh

```diff
@@ -1,28 +1,37 @@
 #!/bin/bash -e -o pipefail
 source ~/utils/utils.sh
 
-createEnvironmentVariable() {
-    local VARIABLE_NAME=$1
-    local JAVA_VERSION=$2
-    if [[ $JAVA_VERSION == "8" ]]; then
-        JAVA_VERSION="1.${JAVA_VERSION}"
+JAVA_TOOLCACHE_PATH=$AGENT_TOOLSDIRECTORY/Java_Adoptium_jdk
+
+installJavaFromAdoptOpenJDK() {
+    local JAVA_VERSION=$1
+
+    javaRelease=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/$JAVA_VERSION/hotspot")
+    archivePath=$(echo $javaRelease | jq -r '[.[] | select(.binary.os=="mac") | .binary.package.link][0]')
```

Comment:
Can we filter by platform once and save result to the variable to avoid code duplication?

Reply:
Fixed it